rpc/src/fetch.rs

```diff
@@ -1,58 +1,64 @@
 use alloy_primitives::B256;
-/// Fetch data from related Portal networks
+use serde::{Deserialize, Serialize};
 use serde_json::Value;
 use tokio::sync::mpsc;
 
 use ethportal_api::{
     types::{
         execution::{block_body::BlockBody, header::Header},
-        jsonrpc::{endpoints::HistoryEndpoint, request::HistoryJsonRpcRequest},
+        history::ContentInfo,
+        jsonrpc::{
+            endpoints::HistoryEndpoint,
+            request::{HistoryJsonRpcRequest, JsonRpcRequest},
+        },
+        query_trace::QueryTrace,
     },
-    utils::bytes::hex_decode,
-    ContentValue, HistoryContentKey, HistoryContentValue,
+    HistoryContentKey, HistoryContentValue,
 };
 
-use crate::errors::RpcServeError;
+use crate::{errors::RpcServeError, serde::from_value};
 
-pub async fn proxy_query_to_history_subnet(
-    network: &mpsc::UnboundedSender<HistoryJsonRpcRequest>,
-    endpoint: HistoryEndpoint,
-) -> Result<Value, RpcServeError> {
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct ContentNotFoundError {
```

Comment:
note: Not sure if there is better place to move this structure to. It would be useful when responding with this error (all `trin-{subnetwork}/src/jsonrpc.rs` files).

Reply:
Maybe move this to `rpc/src/errors.rs`?